app/src/main/java/io/neurolab/communication/bluetooth/BluetoothConnectionManager.java

```diff
@@ -0,0 +1,141 @@
+package io.neurolab.communication.bluetooth;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothSocket;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Set;
+import java.util.UUID;
+
+public class BluetoothConnectionManager {
+
+    // Class Variables
+    private static BluetoothConnectionManager bluetoothConnectionManager;
+
+    // The device can have three possible states (mutually exclusive and exhaustive) at any time related to the local bluetooth adapter.
+    public static final int NO_BLUETOOTH_SUPPORT = 1;  // no bluetooth support
+    public static final int BLUETOOTH_ENABLED = 2;     // bluetooth is supported and enabled
+    public static final int BLUETOOTH_DISABLED = 3;    // bluetooth is supported but disabled
+
+    private static final String DEVICE_MAC_ADDRESS = "xyz";    // TODO: Will be changed later for the headset
+    private static final String DEVICE_NAME = "OPPO A5";      // TODO: Will be changed later for the headset
+    private static final UUID port_uuid = UUID.fromString("00001101-0000-1000-8000-00805F9B34FB");       // TODO: Will be changed for the headset
+
+    // Instance Fields
+    private BluetoothAdapter bluetoothAdapter;    // The local device bluetooth adapter
+
+    private BluetoothSocket bluetoothSocket;
+    private OutputStream outputStream;
+    private InputStream inputStream;
+
+    private BluetoothConnectionManager(){
+        bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
+    }
+
+    public static BluetoothConnectionManager getInstance(){
+        if(bluetoothConnectionManager == null){
+            bluetoothConnectionManager = new BluetoothConnectionManager();
+        }
+        return bluetoothConnectionManager;
+    }
+
+    /*
+    This method checks the current bluetooth state and returns any one of the three states
+    NO_BLUETOOTH_SUPPORT, BLUETOOTH_ENABLED, BLUETOOTH_DISABLED
+     */
+    public int checkBluetoothState(){
+        if(bluetoothAdapter == null)
+            return NO_BLUETOOTH_SUPPORT;
+        else if(!bluetoothAdapter.isEnabled())
+            return BLUETOOTH_DISABLED;
+        else
+            return BLUETOOTH_ENABLED;
+    }
+
+    /*
+    This method checks if the bluetooth adapter is paired with the headset device and updates the
+    bluetoothDevice instance field.
+     */
+    public BluetoothDevice returnPairedHeadset(){
```

Comment:
Does it scan for available bluetooth devices and let user select a device?

Reply:
@CloudyPadmal Currently it doesn't. It lets the user know if they have a headset connected or not. If not would prompt the user to connect to the headset. Would add the scanning functionality as well. Just waiting for the module to arrive to make some final changes. Thanks